src/builtin/filters/array.ts

```diff
@@ -40,7 +41,9 @@ export function slice<T> (v: T[], begin: number, length = 1): T[] {
 export function where<T extends object> (this: FilterImpl, arr: T[], property: string, expected?: any): T[] {
   return toArray(arr).filter(obj => {
     const value = this.context.getFromScope(obj, String(property).split('.'))
-    return expected === undefined ? isTruthy(value, this.context) : value === expected
+    if (expected === undefined) return isTruthy(value, this.context)
+    if (isComparable(expected)) return expected.equals(value)
+    return value === expected
```

Comment:
Maybe we'll need a util `equals()` to do this. I'll check whether there's another case for this.